mne/viz/ica.py

```diff
@@ -130,6 +131,159 @@ def plot_ica_sources(ica, inst, picks=None, exclude=None, start=None,
+                                   for x in axes])
+        if not np.all(is_correct_type):
+            first_bad = np.where(is_correct_type == False)[0][0]
+            raise ValueError('axes must be a list of matplotlib axes objects'
+                             ', while one of the list elements is %s.' %
+                             type(axes[first_bad]))
+        if not len(axes) == 5:
+            raise ValueError('axes must be a list of length 5, while the'
+                             ' length is %d' % len(axes))
+
+    # calculations
+    # ------------
+    if isinstance(inst, _BaseRaw):
+        # calculate segment length that gives us about 250
+        # segments and then break up continuous signal into segments
+        from ..epochs import segment_raw
+        plot_line_at_zero = False
+        seg_len = np.arange(0.5, 10., 0.5)
+        nice_len = inst._data.shape[1] / (250. * inst.info['sfreq'])
+        seg_len = seg_len[np.argmin(np.abs(seg_len - nice_len))]
+        inst = segment_raw(inst, segment_length=seg_len, verbose=False)
+    else:
+        plot_line_at_zero = True
+
+    topo_data = np.dot(ica.mixing_matrix_[:, picks].T,
+                       ica.pca_components_[:ica.n_components_])
+
+    src = ica.get_sources(inst)
+    ica_data = src.get_data()[:, picks, :]
+    ica_data = np.swapaxes(ica_data, 0, 1)
+    evoked = src.average(picks)
+    # smooth_data is kept in a separate variable for future
+    # interactive changing of sigma
+    smooth_data = ndimage.gaussian_filter1d(ica_data[comp_idx],
+                                            sigma=1.5, axis=0)
+    vmin, vmax = _setup_vmin_vmax(smooth_data, None, None)
+
+    # spectrum
+    psds, freqs = psd_multitaper(src, picks=picks)
+    psds = psds[:, comp_idx, :]
+    psds_mean = psds.mean(axis=0)
+    diffs = psds - psds_mean
+    # the distribution of power for each frequency bin is highly
+    # skewed so we calculate std for values below and above average
+    # separately - this is used for fill_between shade
+    pos_sd = [np.sqrt((d[d > 0] ** 2).mean(axis=0)) for d in diffs.T]
+    neg_sd = [np.sqrt((d[d < 0] ** 2).mean(axis=0)) for d in diffs.T]
+
+    # epoch variance
+    epoch_var = np.var(ica_data[comp_idx], axis=1)
+
+    # plotting
+    # --------
+    # FIX topoplot - currently works only for eeg
+    _, pos, merge_grads, names, _ = _prepare_topo_plot(ica, 'eeg', None)
+    plot_topomap(topo_data[comp_idx, :].ravel(), pos, axes=axes[0], show=False)
+    # image
+    img_extent = [1e3 * src.times[0], 1e3 * src.times[-1], 0, len(smooth_data)]
+    axes[1].imshow(smooth_data, extent=img_extent, aspect='auto',
+                   origin='lower', interpolation='nearest',
+                   vmin=vmin, vmax=vmax)
+    # erp
+    axes[2].plot(1e3 * evoked.times, evoked.data[comp_idx].ravel())
+    # spectrum
+    axes[3].plot(freqs, psds_mean, color='k')
+    axes[3].fill_between(freqs, psds_mean - neg_sd, psds_mean + pos_sd,
```

Comment:
I am definitelly going to add labels to the axes (will be useful for interactive stuff), but I can change the variable names here too.